### PR TITLE
Add system prompts for LLMs; system prompts for tool using models.

### PR DESCRIPTION
**Pull request by Jeadie**
## 🗣 Description
 - Create `ChatWrapper` to standardise system prompt injection and tracing in chat models.
 - Add system prompt parameter for LLMs. Enabled via `params.system_prompt`. 
 - For LLMs with `spice_tools` enabled, a system prompt is added to improve tool use. 1.  The datasets available, 2. The tools available.

### Example
Before
```json
{
  "messages": [
    {
      "role": "user",
      "content": "use the vector search tool to find CVEs with quic?"
    }
  ],
  "model": "openai-with-spice"
}
```

After
```json
{
  "messages": [
    {
      "role": "system",
      "content": "Write everything in Haiku like a pirate"
    },
    {
      "role": "system",
      "content": "You have access to a runtime. The following datasets are available in the runtime: spice.nginx.cve. You have access to the following runtime tools: document_similarity, table_schema, sql, list_datasets."
    },
    {
      "role": "user",
      "content": "use the vector search tool to find CVEs with quic?"
    }
  ],
```

With spicepod
```yaml
models:
  - name: openai-with-spice
    from: openai/gpt-4o
    params:
      spice_tools: auto
      openai_api_key: ${ secrets:SPICE_OPENAI_API_KEY }
      system_prompt: |
        Write everything in Haiku like a pirate
```

### CODE REVIEW

File: crates/runtime/src/model/chat.rs [hunk 1]
```diff
@@ -137,7 +148,124 @@ pub fn construct_model<S: ::std::hash::BuildHasher>(
...
+    /// ```rust
+    /// tracing::info!(target: "task_history", truncated_output = %chat_output)
+    /// ```
+    async fn chat_stream(
+        &self,
+        req: CreateChatCompletionRequest,
+    ) -> Result<ChatCompletionResponseStream, OpenAIError> {
+        let span = tracing::span!(target: "task_history", tracing::Level::INFO, "ai_completion", stream=true, model = self.model_name, input = %serde_json::to_string(&req).unwrap_or_default());
+        span.in_scope(
+            || tracing::info!(name: "labels", target: "task_history", model = %req.model),
```
Comment (phillipleblanc) [inline crates/runtime/src/model/chat.rs]: This is redundant if you are adding the `model` to the span directly

File: crates/runtime/src/model/chat.rs [hunk 2]
```diff
@@ -137,7 +148,124 @@ pub fn construct_model<S: ::std::hash::BuildHasher>(
...
+        );
+
+        match self
+            .chat
+            .chat_stream(self.prepare_req(req)?)
+            .instrument(span)
+            .await
+        {
+            Ok(resp) => Ok(resp),
```
Comment (phillipleblanc) [inline crates/runtime/src/model/chat.rs]: You will likely need to instrument the stream itself as well, i.e.:

```rust
Ok(resp) => Ok(resp.instrument(span))
```

File: crates/runtime/src/model/chat.rs [hunk 3]
```diff
@@ -137,7 +148,124 @@ pub fn construct_model<S: ::std::hash::BuildHasher>(
...
+    }
+
+    /// Unlike [`ChatWrapper::chat_stream`], this method will instrument the `truncated_output` for the model output.
+    async fn chat_request(
+        &self,
+        req: CreateChatCompletionRequest,
+    ) -> Result<CreateChatCompletionResponse, OpenAIError> {
+        let span = tracing::span!(target: "task_history", tracing::Level::INFO, "ai_completion", stream=false, model = self.model_name, input = %serde_json::to_string(&req).unwrap_or_default());
+        span.in_scope(
+            || tracing::info!(name: "labels", target: "task_history", model = %req.model),
```
Comment (phillipleblanc) [inline crates/runtime/src/model/chat.rs]: Same comment as above, this is redundant if adding it to the span

Comment (phillipleblanc) [inline crates/runtime/src/model/chat.rs]:
```suggestion
            || tracing::info!(target: "task_history", model = %req.model, "labels"),
```

File: crates/telemetry/src/anonymous.rs [hunk 4]
```diff
@@ -77,16 +77,16 @@ pub async fn start(spicepod_name: &str) {
...
         OtelArrowExporter::new(AnonymousTelemetryExporter::new(Arc::clone(&ENDPOINT)).await);
 
-    let periodic_reader = PeriodicReader::builder(oss_telemetry_exporter.clone(), Tokio)
-        .with_interval(Duration::from_secs(TELEMETRY_INTERVAL_SECONDS))
-        .with_timeout(Duration::from_secs(TELEMETRY_TIMEOUT_SECONDS))
-        .build();
+    // let periodic_reader = PeriodicReader::builder(oss_telemetry_exporter.clone(), Tokio)
+    //     .with_interval(Duration::from_secs(TELEMETRY_INTERVAL_SECONDS))
+    //     .with_timeout(Duration::from_secs(TELEMETRY_TIMEOUT_SECONDS))
+    //     .build();
```
Comment (phillipleblanc) [inline crates/telemetry/src/anonymous.rs]: ?

**Review by ewgenius — APPROVED** [review]

**Pull request merged**